ext/storage_sources/s3_store/aws_bucket_conn.cpp

```diff
@@ -0,0 +1,119 @@
+#include <aws/core/Aws.h>
+#include <aws/s3-crt/S3CrtClient.h>
+#include <aws/s3-crt/model/DeleteObjectRequest.h>
+#include <aws/s3-crt/model/ListObjectsRequest.h>
+#include <aws/s3-crt/model/Object.h>
+#include <aws/s3-crt/model/PutObjectRequest.h>
+#include "aws_bucket_conn.h"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+/*
+ * list_buckets --
+ *     Return list of buckets from AWS account.
```

Comment:
We will need to adapt the description depending on what we choose to do.

Reply:
Hey Etienne, I am not sure what you mean by adapting description? Our list bucket function we are just returning a list of buckets from the AWS account.

Reply:
The function returns a boolean, not a list. We should tell the user what it means if the function returns `true` or `false`. I believe if the function returns `true`, the vector passed by reference to the function will contain a list of buckets?

Reply:
Oh yep, very true! Thank you for clarifying! Will address the changes now.